### ISSUE TITLE
Xcode build failed with code 65 in appium in real ios device.

### ISSUE THREAD

**Issue by sarozNep** [appium/appium#9010]
The gist of the log is as follows> 

 Gist with showXcodeLog true>

https://gist.github.com/sarozNep/a7b2d6d386a1e5376a2a1fe82fa4e7e5

**Comment by sarozNep** [comment]
@imurchie  please help me bro !!

**Comment by mykola-mokhnach** [comment]
> [Xcode] No signing certificate "iOS Development" found:  No "iOS Development" signing certificate matching team ID "KY6JN7H5YN" with a private key was found.

You need to fix the development signature. closed as duplicate

**Comment by sarozNep** [comment]
How to do that ?

**Comment by mykola-mokhnach** [comment]
https://github.com/appium/appium-xcuitest-driver/blob/master/docs/real-device-config.md

**Comment by sarozNep** [comment]
i have done the same as this ..... still having problem

**Comment by mykola-mokhnach** [comment]
Something is still wrong then, because otherwise WDA source should be compiled successfully.

**Comment by sarozNep** [comment]
I haven't done Basic (manual) configuration .. is that necessary ?

**Comment by imurchie** [comment]
You need to go into Xcode and get a signing certificate for ios development.

**Comment by sarozNep** [comment]
i have that certificate too. But still having a problem

**Comment by imurchie** [comment]
Well, the xcodebuild system does not recognize the certificate.

Run xcodebuild itself and see what it says:
```
cd /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent

xcodebuild build-for-testing test-without-building -project /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 -configuration Debug IPHONEOS_DEPLOYMENT_TARGET=10.0 -xcconfig /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-8v3dli.mvrhpvi/appium-temp.xcconfig
```

**Comment by sarozNep** [comment]
V0od0ochild:~ sd$ cd /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent
V0od0ochild:WebDriverAgent sd$ 
V0od0ochild:WebDriverAgent sd$ xcodebuild build-for-testing test-without-building -project /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 -configuration Debug IPHONEOS_DEPLOYMENT_TARGET=10.0 -xcconfig /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-8v3dli.mvrhpvi/appium-temp.xcconfig
Build settings from command line:
    IPHONEOS_DEPLOYMENT_TARGET = 10.0

Build settings from configuration file '/var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-8v3dli.mvrhpvi/appium-temp.xcconfig':
    CODE_SIGN_IDENTITY = iPhone Developer
    DEVELOPMENT_TEAM = KY6JN7H5YN



xcodebuild: error: Unable to find a destination matching the provided destination specifier:
		{ id:2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 }

	Unsupported device specifier option.
	The device “My Mac” does not support the following options: id
	Please supply only supported device specifier options.

	Available destinations for the "WebDriverAgentRunner" scheme:
		{ platform:iOS Simulator, id:FD9FBA01-F235-40E6-BAA8-37F002375C12, OS:10.3, name:iPad Air }
		{ platform:iOS Simulator, id:EB3C5E88-3E79-47E5-B085-C1E43F3490E5, OS:10.3, name:iPad Air 2 }
		{ platform:iOS Simulator, id:FB4329E5-6E5C-4BE1-B480-2FFF4911F975, OS:10.3, name:iPad Pro (9.7 inch) }
		{ platform:iOS Simulator, id:0A4AB87D-D36B-4DF1-B5A2-00682F149F2A, OS:10.3, name:iPad Pro (12.9 inch) }
		{ platform:iOS Simulator, id:B4ADA368-F654-41D0-9BE3-95485CEE4E46, OS:10.3, name:iPhone 5 }
		{ platform:iOS Simulator, id:DACDC209-66B6-413E-9D01-70D4DED576F6, OS:10.3, name:iPhone 5s }
		{ platform:iOS Simulator, id:0E64EACE-6064-4F73-B459-425F3F61962A, OS:10.3, name:iPhone 6 }
		{ platform:iOS Simulator, id:3298508E-9AD2-46DD-8804-AD8EE91DB0DC, OS:10.3, name:iPhone 6 Plus }
		{ platform:iOS Simulator, id:0617B6B6-1855-41BE-883F-107590232231, OS:10.3, name:iPhone 6s }
		{ platform:iOS Simulator, id:519B35B5-206A-43F7-9B98-428EC54CA2C0, OS:10.3, name:iPhone 6s Plus }
		{ platform:iOS Simulator, id:144E902D-71BD-460B-8FFF-2F9EE826289F, OS:10.3, name:iPhone 7 }
		{ platform:iOS Simulator, id:62F0F124-2066-49A9-B051-ED0B130EABDC, OS:10.3, name:iPhone 7 Plus }
		{ platform:iOS Simulator, id:BC1FF285-DE6E-4D93-A48B-CD83AFD900AD, OS:10.3, name:iPhone SE }

	Ineligible destinations for the "WebDriverAgentRunner" scheme:
		{ platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Generic iOS Device }
		{ platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Generic iOS Simulator Device }

**Comment by sarozNep** [comment]
@imurchie 
SEE THE RESULT ;

V0od0ochild:WebDriverAgent sd$ xcodebuild build-for-testing test-without-building -project /Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -
destination id=2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 -configuration Debug IPHONEOS_DEPLOYMENT_TARGET=10.0 -xcconfig /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-8v3dli.mvrhpvi/appium-temp.xcconfig
Build settings from command line:
    IPHONEOS_DEPLOYMENT_TARGET = 10.0

Build settings from configuration file '/var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-8v3dli.mvrhpvi/appium-temp.xcconfig':
    CODE_SIGN_IDENTITY = iPhone Developer
    DEVELOPMENT_TEAM = KY6JN7H5YN

=== BUILD TARGET WebDriverAgentLib OF PROJECT WebDriverAgent WITH CONFIGURATION Debug ===

Check dependencies
No signing certificate "iOS Development" found:  No "iOS Development" signing certificate matching team ID "KY6JN7H5YN" with a private key was found.

** TEST BUILD FAILED **


The following build commands failed:
	Check dependencies
(1 failure)
2017-08-11 13:59:07.476 xcodebuild[3573:73030]  IDETestOperationsObserverDebug: Writing diagnostic log for test session to:
/Users/sd/Library/Developer/Xcode/DerivedData/WebDriverAgent-dikkwtrisltbeobjmfvpthwwekvs/Logs/Test/EA4DBA36-DA4D-4C32-ACBE-4E5BCC3826FE/Session-WebDriverAgentRunner-2017-08-11_135907-uhZowJ.log
2017-08-11 13:59:07.476 xcodebuild[3573:73027] [MT] IDETestOperationsObserverDebug: (E1DFE37C-6DDF-49ED-B9E3-3872F1144A4E) Beginning test session WebDriverAgentRunner-E1DFE37C-6DDF-49ED-B9E3-3872F1144A4E at 2017-08-11 13:59:07.476 with Xcode 8E2002 on target <DVTiOSDevice: 0x7fd85940e970> {
		deviceSerialNumber:         DNPMFNKCFFDR
		identifier:                 2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2
		deviceClass:                iPhone
		deviceName:                 Domingos’s iPhone
		deviceIdentifier:           2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2
		productVersion:             10.0.2
		buildVersion:               14A456
		deviceSoftwareVersion:      10.0.2 (14A456)
		deviceArchitecture:         arm64
		deviceTotalCapacity:        12263628800
		deviceAvailableCapacity:    9806032896
		deviceIsTransient:          NO
		ignored:                    NO
		deviceIsBusy:               NO
		deviceIsActivated:          YES
		deviceActivationState:      Activated
		isPasscodeLocked:           NO
		deviceType:                 <DVTDeviceType:0x7fd858cc9340 Xcode.DeviceType.iPhone>
		supportedDeviceFamilies:    (
    1
)
		applications:              (null)
		provisioningProfiles:      (null)
		activityProgress:          -2
		activityTitle:             
		hasInternalSupport:        NO
		isSupportedOS:             YES
		developerDiskMountError:   (null)
(null)
	bootArgs:                  <unavailable>
		} (10.0.2 (14A456))
2017-08-11 13:59:07.496 xcodebuild[3573:73027] Error Domain=NSCocoaErrorDomain Code=260 "The file “WebDriverAgentRunner-Runner.app” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/Build/Products/Debug-iphoneos/WebDriverAgentRunner-Runner.app, NSUnderlyingError=0x7fd859674740 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
2017-08-11 13:59:07.496 xcodebuild[3573:73027] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}
** TEST EXECUTE FAILED **

**Comment by imurchie** [comment]
So, the same as when run in Appium. The error is pretty clear.

Here's some stuff from Google:
https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/Troubleshooting/Troubleshooting.html
https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppDistributionGuide/MaintainingCertificates/MaintainingCertificates.html

**Comment by sarozNep** [comment]
`what should i do now?

**Comment by imurchie** [comment]
I would open `/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/WebDriverAgent/WebDriverAgent.xcodeproj` in Xcode, and see what the signing status is there.

**Comment by sarozNep** [comment]
I have opened it in xcode and when i build xcode with signing certificate and team .. build is success .... what maybe the issue :/

**Comment by sarozNep** [comment]
After i deleted the existing app in device  ! i have run appium ....

Here is another error ...

[Appium] Welcome to Appium v1.6.5
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> GET /wd/hub/sessions {}
[MJSONWP] Calling AppiumDriver.getSessions() with args: []
[MJSONWP] Responding to client with driver.getSessions() result: []
[HTTP] <-- GET /wd/hub/sessions 200 10 ms - 40 
[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"app":"/Users/sd/Desktop/Allian-iOS.DEV.ipa","bundleId":"com.muzzley.ios.allian.DEV","deviceName":"iPhone 5s","noReset":true,"platformName":"iOS","platformVersion":"10.0.2","showXcodeLog":true,"udid":"2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2","xcodeOrgId":"KY6JN7H5YN","xcodeSigningId":"iPhone Developer","newCommandTimeout":0,"connectHardwareKeyboard":true}}
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"/Users/sd/Desktop/Allianz-iOS.DEV.ipa","bundleId":"com.sd.ios.allian.DEV","deviceName":"iPhone 5s","noReset":true,"platformName":"iOS","platformVersion":"10.0.2","showXcodeLog":true,"udid":"2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2","xcodeOrgId":"KY6JN7H5YN","xcodeSigningId":"iPhone Developer","newCommandTimeout":0,"connectHardwareKeyboard":true},null,null]
[BaseDriver] Event 'newSessionRequested' logged at 1502459234024 (14:47:14 GMT+0100 (WEST))
[Appium] Requested iOS support with version >= 10, using XCUITest driver instead of UIAutomation-based driver, since the latter is unsupported on iOS 10 and up.
[Appium] Creating new XCUITestDriver (v2.30.0) session
[Appium] Capabilities:
[Appium]   app: '/Users/sd/Desktop/Allian-iOS.DEV.ipa'
[Appium]   bundleId: 'com.sd.ios.allian.smarthome.DEV'
[Appium]   deviceName: 'iPhone 5s'
[Appium]   noReset: true
[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '10.0.2'
[Appium]   showXcodeLog: true
[Appium]   udid: '2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2'
[Appium]   xcodeOrgId: 'KY6JN7H5YN'
[Appium]   xcodeSigningId: 'iPhone Developer'
[Appium]   newCommandTimeout: 0
[Appium]   connectHardwareKeyboard: true
[XCUITest] XCUITestDriver version: 2.30.0
[BaseDriver] Session created with session id: 365b0247-ffb9-4fe2-8076-99f09eac5485
[XCUITest] Xcode version set to '8.3.2' (tools v8.3.2.0.1.1492020469)
[XCUITest] iOS SDK Version set to '10.3'
[BaseDriver] Event 'xcodeDetailsRetrieved' logged at 1502459234188 (14:47:14 GMT+0100 (WEST))
[XCUITest] Available devices: 2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2
[XCUITest] Creating iDevice object with udid '2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2'
[XCUITest] Determining device to run tests on: udid: '2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2', real device: true
[BaseDriver] Using local app '/Users/sd/Desktop/Allian-iOS.DEV.ipa'
[BaseDriver] Copying local zip to tmp dir
[BaseDriver] Unzipping /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/appium-app.zip
[BaseDriver] Testing zip archive: /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/appium-app.zip
[BaseDriver] Unzipped local app to '/var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allian-iOS.DEV.app'
[BaseDriver] Event 'appConfigured' logged at 1502459241039 (14:47:21 GMT+0100 (WEST))
[XCUITest] Checking whether app '/var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allianz-iOS.DEV.app' is actually present on file system
[XCUITest] App is present
[BaseDriver] Event 'resetStarted' logged at 1502459241043 (14:47:21 GMT+0100 (WEST))
[XCUITest] Reset: fullReset not set. Leaving as is
[BaseDriver] Event 'resetComplete' logged at 1502459241044 (14:47:21 GMT+0100 (WEST))
[iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog
[iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'
[BaseDriver] Event 'logCaptureStarted' logged at 1502459241172 (14:47:21 GMT+0100 (WEST))
[XCUITest] Setting up real device
[XCUITest] Installing **'/var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allian-iOS.DEV.app'** on device with UUID '2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2'...
[XCUITest] Error: Could not install app: 'Command 'ios-deploy --id 2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 --uninstall --bundle /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allianz-iOS.DEV.app' exited with code 253'
    at IOSDeploy.install$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:38:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at process._tickCallback (internal/process/next_tick.js:103:7)
 Error: Could not install app: 'Command 'ios-deploy --id 2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 --uninstall --bundle /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allianz-iOS.DEV.app' exited with code 253'
    at IOSDeploy.install$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:38:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at process._tickCallback (internal/process/next_tick.js:103:7)
[XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[iOSLog] Stopping iOS log capture
[MJSONWP] **Encountered internal error running command: Error: Could not install app: 'Command 'ios-deploy --id 2a76e49bb8c9ad5a983e007ea8d2ded23f7bd5c2 --uninstall --bundle /var/folders/gm/49c1hvk93b91915wrh1lr95m0000gp/T/2017711-557-16b5nv5.0wbw23ayvi/Payload/Allian-iOS.DEV.app'** exited with code 253'
    at IOSDeploy.install$ (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/appium-xcuitest-driver/lib/ios-deploy.js:38:13)
    at tryCatch (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium.app/Contents/Resources/app/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at process._tickCallback (internal/process/next_tick.js:103:7)
[HTTP] <-- POST /wd/hub/session 500 16574 ms - 378
